emscripten.py

```diff
@@ -1540,16 +1552,24 @@ def create_receiving(function_table_data, function_tables_defs, exported_impleme
 
   if not shared.Settings.SWAPPABLE_ASM_MODULE:
     if shared.Settings.DECLARE_ASM_MODULE_EXPORTS:
-      receiving += ';\n'.join(['var ' + s + ' = Module["' + s + '"] = asm["' + s + '"]' for s in module_exports])
+      if shared.Settings.WASM and shared.Settings.MINIMAL_RUNTIME:
+        receiving += '\n'.join([s + ' = asm["' + s + '"];' for s in module_exports]) + '\n'
+      else:
+        if shared.Settings.MINIMAL_RUNTIME:
+          receiving += '\n'.join(['var ' + s + ' = asm["' + s + '"];' for s in module_exports]) + '\n'
+        else:
+          receiving += '\n'.join(['var ' + s + ' = Module["' + s + '"] = asm["' + s + '"];' for s in module_exports]) + '\n'
       # TODO: Instead of the above line, we would like to use the two lines below for smaller size version of exports; but currently JS optimizer is wired to look for the exact above syntax when analyzing
       # exports.
 #      receiving += ''.join(['var ' + s + ' = asm["' + s + '"];\n' for s in module_exports])
 #      receiving += 'for(var module_exported_function in asm) Module[module_exported_function] = asm[module_exported_function];\n'
     else:
-      receiving += 'for(var i in asm) this[i] = Module[i] = asm[i];\n'
+      if shared.Settings.MINIMAL_RUNTIME:
+        receiving += 'for(var i in asm) this[i] = asm[i];\n'
+      else:
+        receiving += 'for(var i in asm) this[i] = Module[i] = asm[i];\n'
   else:
-    receiving += 'Module["asm"] = asm;\n' + ';\n'.join(['var ' + s + ' = Module["' + s + '"] = function() {' + runtime_assertions + '  return Module["asm"]["' + s + '"].apply(null, arguments) }' for s in module_exports])
-  receiving += ';\n'
+    receiving += 'Module["asm"] = asm;\n' + '\n'.join(['var ' + s + ' = Module["' + s + '"] = function() {' + runtime_assertions + '  return Module["asm"]["' + s + '"].apply(null, arguments) };' for s in module_exports]) + '\n'
```

Comment:
In MINIMAL_RUNTIME we do not unconditionally export everything to Module. (Instead, in the future we should have a more explicit model that generates an export object from only those symbols that need it. Currently exports are not supported)